pkg/api/http_server.go

```diff
@@ -841,3 +787,141 @@ func (hs *HTTPServer) getDefaultCiphers(tlsVersion uint16, protocol string) []ui
 	}
 	return nil
 }
+
+func (hs *HTTPServer) readCertificates() (*tls.Certificate, error) {
+	if hs.Cfg.CertFile == "" {
+		return nil, errors.New("cert_file cannot be empty when using HTTPS")
+	}
+
+	if hs.Cfg.KeyFile == "" {
+		return nil, errors.New("cert_key cannot be empty when using HTTPS")
+	}
+
+	if _, err := os.Stat(hs.Cfg.CertFile); os.IsNotExist(err) {
+		return nil, fmt.Errorf(`cannot find SSL cert_file at %q`, hs.Cfg.CertFile)
+	}
+
+	if _, err := os.Stat(hs.Cfg.KeyFile); os.IsNotExist(err) {
+		return nil, fmt.Errorf(`cannot find SSL key_file at %q`, hs.Cfg.KeyFile)
+	}
+
```

Comment:
Let's move these back up into tlsCertificates above line 567 so that the user gets these friendly errors at startup if the configuration is invalid, rather than getting the raw os.Stat error from updateMtimeOfServerCerts

Reply:
will move updateMtimeOfServreCerts after the readCertificates. So these errors come properly